python/cudf/cudf/core/dataframe.py

```diff
@@ -253,7 +253,9 @@ def _init_from_dict_like(self, data, index=None, columns=None):
                 if is_scalar(data[col_name]):
                     num_rows = num_rows or 1
                 else:
-                    data[col_name] = column.as_column(data[col_name])
+                    data[col_name] = column.as_column(
+                        data[col_name], nan_as_null=True
+                    )
```

Comment:
This now looks like if I pass a dict of Columns that it will convert any `NaN` values in my float columns to nulls erroneously.

Reply:
Removing this was causing the issue with `NAT` in date time column, will dig deeper.